src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -403,7 +404,12 @@ private SnapshotProject retrieveSnapshotProject(String sql, MapSqlParameterSourc
         // add the filter to the clause to get the actual items
         DaoUtils.addFilterClause(filter, params, whereClauses);
         String whereSql = "";
-        if (!whereClauses.isEmpty()) {
+        String joinSql = "";
+        if (!whereClauses.isEmpty() || !StringUtils.isEmpty(datasetId)) { //TODO can whereClauses be empty w line 402?
```

Comment:
I agree, it can't be empty anymore. Rewriting this code to remove the empty checks should simplify things.